Product/Production/Services/CORE_X12DocumentSubmissionCore/src/test/java/gov/hhs/fha/nhinc/corex12/docsubmission/realtime/inbound/PassthroughInboundCORE_X12DSRealTimeTest.java

```diff
@@ -0,0 +1,111 @@
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import java.util.Properties;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeResponse;
+import org.junit.Test;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class PassthroughInboundCORE_X12DSRealTimeTest {
+
+    private final AuditEJBLoggerImpl mockEJBLogger = mock(AuditEJBLoggerImpl.class);
+
+    @Test
+    public void auditLoggingOnForInboundRealTime() {
+        AdapterCORE_X12DSRealTimeProxyObjectFactory mockFactory = mock(AdapterCORE_X12DSRealTimeProxyObjectFactory.class);
+        PassthroughInboundCORE_X12DSRealTime realTimeX12
+            = new PassthroughInboundCORE_X12DSRealTime(mockFactory, getAuditLogger(true));
+        COREEnvelopeRealTimeRequest request = new COREEnvelopeRealTimeRequest();
+        AssertionType assertion = new AssertionType();
+        Properties webContextProperties = new Properties();
+        AdapterCORE_X12DSRealTimeProxy mockAdapterProxy = mock(AdapterCORE_X12DSRealTimeProxy.class);
+        when(mockFactory.getAdapterCORE_X12DocSubmissionProxy()).thenReturn(mockAdapterProxy);
+        COREEnvelopeRealTimeResponse expectedResponse = new COREEnvelopeRealTimeResponse();
+        when(mockAdapterProxy.realTimeTransaction(request, assertion)).thenReturn(expectedResponse);
+        COREEnvelopeRealTimeResponse actualResponse = realTimeX12.realTimeTransaction(request, assertion,
+            webContextProperties);
+        verify(mockEJBLogger).auditResponseMessage(eq(request), eq(actualResponse), eq(assertion),
+            isNull(NhinTargetSystemType.class), eq(NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION),
+            eq(NhincConstants.AUDIT_LOG_NHIN_INTERFACE), eq(Boolean.FALSE), eq(webContextProperties),
+            eq(NhincConstants.CORE_X12DS_REALTIME_SERVICE_NAME), any(COREX12RealTimeAuditTransforms.class));
+    }
+
+    @Test
+    public void auditLoggingOffForInboundRealTime() {
+        AdapterCORE_X12DSRealTimeProxyObjectFactory mockFactory = mock(AdapterCORE_X12DSRealTimeProxyObjectFactory.class);
+        PassthroughInboundCORE_X12DSRealTime realTimeX12
+            = new PassthroughInboundCORE_X12DSRealTime(mockFactory, getAuditLogger(false));
+        COREEnvelopeRealTimeRequest request = new COREEnvelopeRealTimeRequest();
+        AssertionType assertion = new AssertionType();
+        Properties webContextProperties = new Properties();
+        AdapterCORE_X12DSRealTimeProxy mockAdapterProxy = mock(AdapterCORE_X12DSRealTimeProxy.class);
+        when(mockFactory.getAdapterCORE_X12DocSubmissionProxy()).thenReturn(mockAdapterProxy);
```

Comment:
These variables are used in both methods. Can they be declared as class variables?